### PR TITLE
chore: create yaml schemas and types for ingestion-trigger(MAPCO-1783, MAPCO-4135)

### PR DESCRIPTION
**Pull request by almog8k**

### CODE REVIEW

File: src/models/raster/ingestion/ingestionLayer.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import { LayerMetadata } from '../../layerMetadata';
```
Comment (asafMasa) [inline src/models/raster/ingestion/ingestionLayer.ts:1]: maybe rename the file to something more precise such as "ingestionTypes"

File: src/yaml/ingestionTrigger/inputFiles/rasterLayerInputFiles.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+          type: string
+          example: 'string'
+          description: layer source directory
+        fileNames:
+          type: array
+          items:
+            type: string
+            pattern: ^.+\.[Gg][Pp][Kk][Gg]$
+            description: layer source files
+          example: ['example.tif']
```
Comment (asafMasa) [inline src/yaml/ingestionTrigger/inputFiles/rasterLayerInputFiles.yaml:21]: change example to ['example.gpkg']

File: src/yaml/ingestionTrigger/partData/rasterLayerPartData.yaml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+              type: string
+              description: Oldest imaging date (UTC)
+              format: date-time
+            imagingTimeEndUTC:
+              type: string
+              description: Latest imaging date (UTC)
+              format: date-time
+            resolutionDegree:
+              type: number
+              minimum: 9.e-8
```
Comment (asafMasa) [inline src/yaml/ingestionTrigger/partData/rasterLayerPartData.yaml:38]: please update by the following:
Min res Degree to pixel: 0.703125
Max res Degree to pixel: 1.67638e-7
Min res Meter to pixel: 78273
Max res Meter to pixel: 0.0185

Update also in decorators

File: src/yaml/ingestionTrigger/partData/rasterLayerPartData.yaml [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+                type: string
+              description: cities included in the layer
+            classification:
+              type: string
+              description: Permitted roles
+            description:
+              type: string
+            geometry:
+              $ref: '../../geojson.yaml#/components/schemas/Geometry'
+              example: ['example.tif']
```
Comment (asafMasa) [inline src/yaml/ingestionTrigger/partData/rasterLayerPartData.yaml:82]: change example ['example.gpkg']

File: src/yaml/ingestionTrigger/rasterLayerExample.yaml [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+                - - 34.86824157112912
+                  - 32.29430955805424
+                - - 34.86824157112912
+                  - 32.30543192283443
+                - - 34.85149443279957
+                  - 32.30543192283443
+        inputFiles:            
+          originDirectory: string
+          fileNames:
+            - example.tif
```
Comment (asafMasa) [inline src/yaml/ingestionTrigger/rasterLayerExample.yaml:54]: change example 'example.gpkg' in all yamls

File: src/yaml/ingestionTrigger/rasterLayerExample.yaml [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
```
Comment (asafMasa) [inline src/yaml/ingestionTrigger/rasterLayerExample.yaml:95]: change example 'example.gpkg'

File: src/yaml/ingestionTrigger/rasterLayerExample.yaml [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+          region: string
+          classification: string
+          producerName: string
+          scale: 100000000
+        partData:
+          - id: string
+            name: string
+            imagingTimeBeginUTC: "2024-01-28T13:47:43.427Z"
+            imagingTimeEndUTC: "2024-01-28T13:47:43.427Z"
+            resolutionDegree: 0.072
```
Comment (asafMasa) [inline src/yaml/ingestionTrigger/rasterLayerExample.yaml:27]: in all examples change to min of 0.703125

File: src/yaml/ingestionTrigger/rasterLayerExample.yaml [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+          description: string
+          region: string
+          classification: string
+          scale: 100000000
+        partData:
+          - id: string
+            name: string
+            imagingTimeBeginUTC: "2024-01-28T13:47:43.427Z"
+            imagingTimeEndUTC: "2024-01-28T13:47:43.427Z"
+            resolutionDegree: 0.072
```
Comment (asafMasa) [inline src/yaml/ingestionTrigger/rasterLayerExample.yaml:68]: change to min 0.703125 in all yamls

File: src/yaml/ingestionTrigger/responses/ingestionTriggerResponses.yaml [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+            $ref: '../../geojson.yaml#/components/schemas/Geometry'
+        required:
+          - crs
+          - fileFormat
+          - pixelSize
+          - footprint
+        example:
+          crs: 4326
+          fileFormat: gpkg
+          pixelSize: 0.072
```
Comment (asafMasa) [inline src/yaml/ingestionTrigger/responses/ingestionTriggerResponses.yaml:28]: min of 0.703125

**Review by asafMasa — CHANGES_REQUESTED** [review]
Good job

File: src/models/raster/ingestion/ingestionLayer.ts [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+import { PolygonPartRecord } from '../../polygonParts';
+
+export type UpdateRasterLayerMetadata = Pick<LayerMetadata, 'productSubType' | 'description' | 'region' | 'classification' | 'scale'>;
+export type NewRasterLayerMetadata = UpdateRasterLayerMetadata &
```
[the remark this answers is not in the document]
Comment (almog8k) [reply src/models/raster/ingestion/ingestionLayer.ts:5]: done

File: src/yaml/ingestionTrigger/partData/rasterLayerPartData.yaml (hunk 3 above)
Comment (almog8k) [reply src/yaml/ingestionTrigger/partData/rasterLayerPartData.yaml:38]: what do you mean by update decorators?

File: src/yaml/ingestionTrigger/rasterLayerExample.yaml (hunk 5 above)
Comment (almog8k) [reply src/yaml/ingestionTrigger/rasterLayerExample.yaml:54]: done

File: src/yaml/ingestionTrigger/rasterLayerExample.yaml (hunk 6 above)
Comment (almog8k) [reply src/yaml/ingestionTrigger/rasterLayerExample.yaml:95]: done

File: src/yaml/ingestionTrigger/rasterLayerExample.yaml (hunk 7 above)
Comment (almog8k) [reply src/yaml/ingestionTrigger/rasterLayerExample.yaml:27]: done

File: src/yaml/ingestionTrigger/rasterLayerExample.yaml (hunk 8 above)
Comment (almog8k) [reply src/yaml/ingestionTrigger/rasterLayerExample.yaml:68]: done

File: src/yaml/ingestionTrigger/inputFiles/rasterLayerInputFiles.yaml (hunk 2 above)
Comment (almog8k) [reply src/yaml/ingestionTrigger/inputFiles/rasterLayerInputFiles.yaml]: done

File: src/models/raster/ingestion/ingestionLayer.ts (hunk 1 above)
Comment (almog8k) [reply src/models/raster/ingestion/ingestionLayer.ts]: done

**Review by asafMasa — APPROVED** [review]

**Review by netanelC — APPROVED** [review]

**Review by alebinson — CHANGES_REQUESTED** [review]
Functions for reflecting decoration metadata are missing

**Review by asafMasa — CHANGES_REQUESTED** [review]

File: src/models/polygonParts/polygonPartRecord.ts [hunk 11]
```diff
@@ -511,7 +511,7 @@ export class PolygonPartRecord implements IPolygonPart {
     ],
   })
   //#endregion
-  public version: string | undefined = undefined;
+  public updatedInVersion: string | undefined = undefined;
```
[the remark this answers is not in the document]
Comment (almog8k) [reply src/models/polygonParts/polygonPartRecord.ts:514]: versionChangedName

File: src/models/polygonParts/polygonPartRecord.ts [hunk 12]
```diff
@@ -135,9 +141,9 @@ export class PolygonPartRecord implements IPolygonPart {
     },
   })
   @inputDataMapping({
-    isCustomLogic: true,
+    isCustomLogic: false,
     dataFile: DataFileType.SHAPE_METADATA,
-    valuePath: '***max(features[].properties.UpdateDate)***',
+    valuePath: 'properties.UpdateDate',
```
[the remark this answers is not in the document]
Comment (almog8k) [reply src/models/polygonParts/polygonPartRecord.ts:146]: https://files.slack.com/files-pri/T01JHEZHFUM-F070KLYR0E4/image.png

File: src/models/polygonParts/polygonPartRecord.ts [hunk 13]
```diff
@@ -99,9 +105,9 @@ export class PolygonPartRecord implements IPolygonPart {
     },
   })
   @inputDataMapping({
-    isCustomLogic: true,
+    isCustomLogic: false,
     dataFile: DataFileType.SHAPE_METADATA,
-    valuePath: '***min(features[].properties.UpdateDate)***',
+    valuePath: 'properties.UpdateDate',
```
[the remark this answers is not in the document]
Comment (almog8k) [reply src/models/polygonParts/polygonPartRecord.ts:110]: image

File: src/yaml/ingestionTrigger/metadata/rasterLayerMetadata.yaml [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+          type: string
+          description: Layer's sub type
+        description:
+          type: string
+          description: layer's description
+        region:
+          type: string
+          description: region of record
+        classification:
+          type: string
```
Comment (asafMasa) [inline src/yaml/ingestionTrigger/metadata/rasterLayerMetadata.yaml:24]: should be defined as enum with the values defined in meta-data file

File: src/yaml/ingestionTrigger/partData/rasterLayerPartData.yaml [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+              type: array
+              items:
+                type: string
+              description: countries included in the layer
+            cities:
+              type: array
+              items:
+                type: string
+              description: cities included in the layer
+            classification:
```
Comment (asafMasa) [inline src/yaml/ingestionTrigger/partData/rasterLayerPartData.yaml:75]: doesn't exist - should be removed

File: src/yaml/ingestionTrigger/partData/rasterLayerPartData.yaml [hunk 16]
```diff
@@ -0,0 +1,82 @@
...
+            - imagingTimeBeginUTC
+            - imagingTimeEndUTC
+            - geometry
+            - resolutionDegree
+            - resolutionMeter
+            - sourceResolutionMeter
+            - horizontalAccuracyCE90
+            - sensors
+          properties:
+            id:
```
Comment (asafMasa) [inline src/yaml/ingestionTrigger/partData/rasterLayerPartData.yaml:21]: check if this should be mandatory, please remove the "pattern" attribute

File: src/yaml/ingestionTrigger/rasterLayerExample.yaml [hunk 17]
```diff
@@ -0,0 +1,95 @@
...
+          productId: string
+          productName: string
+          productType: Orthophoto
+          productSubType: string
+          description: string
+          srs: string
+          srsName: string
+          transparency: TRANSPARENT
+          region: string
+          classification: string
```
Comment (asafMasa) [inline src/yaml/ingestionTrigger/rasterLayerExample.yaml:19]: should be with value as defined in metadata file

File: src/yaml/ingestionTrigger/rasterLayerExample.yaml [hunk 18]
```diff
@@ -0,0 +1,95 @@
...
+  examples:
+    NewRasterLayerExample:
+      value:
+        metadata:
+          productId: string
+          productName: string
+          productType: Orthophoto
+          productSubType: string
+          description: string
+          srs: string
```
Comment (asafMasa) [inline src/yaml/ingestionTrigger/rasterLayerExample.yaml:15]: should be with value as defined in metadata file

File: src/yaml/ingestionTrigger/rasterLayerExample.yaml [hunk 19]
```diff
@@ -0,0 +1,95 @@
...
+          srsName: string
```
Comment (asafMasa) [inline src/yaml/ingestionTrigger/rasterLayerExample.yaml:16]: should be with value as defined in metadata file

File: src/yaml/ingestionTrigger/rasterLayerExample.yaml [hunk 20]
```diff
@@ -0,0 +1,95 @@
...
+          originDirectory: string
+          fileNames:
+            - 'example.gpkg'
+    UpdateRasterLayerExample:
+      value:
+        metadata:
+          productSubType: string
+          description: string
+          region: string
+          classification: string
```
Comment (asafMasa) [inline src/yaml/ingestionTrigger/rasterLayerExample.yaml:61]: should be with value as defined in metadata file

**Review by asafMasa — CHANGES_REQUESTED** [review]

File: src/models/raster/ingestion/ingestionTypes.ts [hunk 21]
```diff
@@ -0,0 +1,72 @@
...
+export type IUpdateRasterLayerMetadata = Pick<LayerMetadata, 'productSubType' | 'description' | 'region' | 'classification' | 'scale'>;
+
+export class UpdateRasterLayerMetadata implements IUpdateRasterLayerMetadata {
+  public productSubType: string | undefined;
+  public description: string | undefined;
+  public region: string[] | undefined;
+  public classification: string | undefined;
+  public scale: number | undefined;
+
+  public constructor() {
```
[the remark this answers is not in the document]
Comment (almog8k) [reply src/models/raster/ingestion/ingestionTypes.ts:13]: the only use of the "UpdateRasterLayerMetadata" class is to extract the Object keys (we are not creating instances from this class)

File: src/models/raster/ingestion/ingestionTypes.ts [hunk 22]
```diff
@@ -0,0 +1,72 @@
...
+  public productId: string | undefined;
+  public producerName: string | undefined;
+  public productType: ProductType | undefined;
+  public srs: string | undefined;
+  public srsName: string | undefined;
+  public transparency: Transparency | undefined;
+  public productName: string | undefined;
+
+  public constructor() {
+    super();
```
[the remark this answers is not in the document]
Comment (almog8k) [reply src/models/raster/ingestion/ingestionTypes.ts:35]: the only use of the "NewRasterLayerMetadata" class is to extract the Object keys (we are not creating instances from this class)

File: src/yaml/ingestionTrigger/partData/rasterLayerPartData.yaml (hunk 15 above)
Comment (almog8k) [reply src/yaml/ingestionTrigger/partData/rasterLayerPartData.yaml:75]: cities?

File: src/yaml/ingestionTrigger/partData/rasterLayerPartData.yaml (hunk 16 above)
Comment (almog8k) [reply src/yaml/ingestionTrigger/partData/rasterLayerPartData.yaml:21]: pattern removed, i'll check with shaziri

File: src/models/polygonParts/polygonPartRecord.ts [hunk 23]
```diff
@@ -76,7 +484,7 @@ export class PolygonPartRecord extends PolygonPart implements IOrmCatalog {
   //#endregion
   public recordId: string | undefined = 'UNKNOWN';
 
-  //#region RECORD: version
+  //#region RECORD: updatedInVersion [Version number of the best layer when it was updated]
   @catalogDB({
     column: {
       name: 'version',
```
[the remark this answers is not in the document]
Comment (almog8k) [reply src/models/polygonParts/polygonPartRecord.ts:490]: done

File: src/yaml/ingestionTrigger/metadata/rasterLayerMetadata.yaml (hunk 14 above)
Comment (almog8k) [reply src/yaml/ingestionTrigger/metadata/rasterLayerMetadata.yaml:24]: reverted

File: src/yaml/ingestionTrigger/partData/rasterLayerPartData.yaml (hunk 16 above)
Comment (almog8k) [reply src/yaml/ingestionTrigger/partData/rasterLayerPartData.yaml:21]: field currently is not required

File: src/yaml/ingestionTrigger/rasterLayerExample.yaml (hunk 17 above)
Comment (almog8k) [reply src/yaml/ingestionTrigger/rasterLayerExample.yaml:19]: done

File: src/yaml/ingestionTrigger/rasterLayerExample.yaml (hunk 18 above)
Comment (almog8k) [reply src/yaml/ingestionTrigger/rasterLayerExample.yaml:15]: done

File: src/yaml/ingestionTrigger/rasterLayerExample.yaml (hunk 19 above)
Comment (almog8k) [reply src/yaml/ingestionTrigger/rasterLayerExample.yaml:16]: done

File: src/yaml/ingestionTrigger/rasterLayerExample.yaml (hunk 20 above)
Comment (almog8k) [reply src/yaml/ingestionTrigger/rasterLayerExample.yaml:61]: done

**Review by asafMasa — APPROVED** [review]

**Review by alebinson — APPROVED** [review]

**Pull request merged**